Diagnostics/Get-NetView.PS1

```diff
@@ -880,12 +879,14 @@ function ChelsioDetail {
         return $null
     }
 
-    $ifName = (Get-NetAdapter -InterfaceIndex $IfIndex).Name
+    $ifName = $Nic.Name
+    $ifIndex = $Nic.ifIndex
     $ifHwInfo = Get-NetAdapterHardwareInfo -Name "$ifName"
     $dirBusName = "BusDev_" + $ifHwInfo.Bus + "_" + $ifHwInfo.Device + "_" + $ifHwInfo.Function
     $dirBus = (Join-Path -Path $dir -ChildPath $dirBusName)
 
-    if (-not (Test-Path $dirBus)) { # run once
+    if (-not $Global:ChelsioRunOnce) { # expensive
```

Comment:
Takes some time to complete, and generates a somewhat large file.

Reply:
ChelsioRunOnce can't be a global.  There may be multiple physical NICs in the system in which case each must have the command executed.  It's "globalness" is with regard to the number of Ports on a given ASIC.   I'd say that if this takes a long time, then use a parallel execution mechanism as already in use in the main function Get-NetView().

Reply:
It also outputs 16 MB of files (~2 MB) compressed. Running it once per "$dirBusName" should be good enough.